addons/addon-base-raas/packages/base-raas-services/lib/data-source/access-strategy/roles-only/__tests__/environment-resource-service.test.js

```diff
@@ -0,0 +1,159 @@
+const JsonSchemaValidationService = require('@aws-ee/base-services/lib/json-schema-validation-service');
+
+const { StudyPolicy } = require('../../../../helpers/iam/study-policy');
+const EnvironmentScService = require('../../../../environment/service-catalog/environment-sc-service');
+const DataSourceBucketService = require('../../../data-source-bucket-service');
+const ResourceUsageService = require('../../../../usage/resource-usage-service');
+const ApplicationRoleService = require('../application-role-service');
+const FilesystemRoleService = require('../filesystem-role-service');
+const EnvironmentResourceService = require('../environment-resource-service');
+
+const createStudy = ({
+  id = 'study-1',
+  category = 'Organization',
+  accountId = '1122334455',
+  awsPartition = 'aws',
+  bucketAccess = 'roles',
+  bucket = 'bucket-1',
+  qualifier = 'swb-IhsKhN8GsLneiis11ujlb8',
+  appRoleArn = 'arn:aws:iam::684277579687:role/swb-IhsKhN8GsLneiis11ujlb8-app-1607537845811',
+  accessType = 'readwrite',
+  envPermission = { read: true, write: true },
+  folder = '/',
+  kmsScope = 'none',
+  region = 'us-east-1',
+} = {}) => ({
+  id,
+  category,
+  accountId,
+  awsPartition,
+  bucketAccess,
+  bucket,
+  qualifier,
+  appRoleArn,
+  accessType,
+  envPermission,
+  folder,
+  kmsScope,
+  region,
+});
+
+const createAdminContext = ({ uid = 'uid-admin' } = {}) => ({
+  principalIdentifier: { uid },
+  principal: { isAdmin: true, userRole: 'admin', status: 'active' },
+});
+
+describe('EnvironmentResourceService', () => {
+  let container;
+  let service;
+
+  beforeEach(async () => {
+    // Initialize services container and register dependencies
+    container = new ServicesContainer();
+
+    container.register('aws', new Aws());
+    container.register('log', new Logger());
+    container.register('lockService', new LockService());
+    container.register('dbService', new DbService());
+    container.register('jsonSchemaValidationService', new JsonSchemaValidationService());
+    container.register('settings', new SettingsService());
+    container.register('pluginRegistryService', new PluginRegistryService());
+    container.register('authorizationService', new AuthService());
+    container.register('auditWriterService', new AuditService());
+    container.register('dataSourceBucketService', new DataSourceBucketService());
+    container.register('environmentScService', new EnvironmentScService());
+    container.register('resourceUsageService', new ResourceUsageService());
+    container.register('roles-only/applicationRoleService', new ApplicationRoleService());
+    container.register('roles-only/filesystemRoleService', new FilesystemRoleService());
+    container.register('roles-only/environmentResourceService', new EnvironmentResourceService());
+    await container.initServices();
+
+    service = await container.find('roles-only/environmentResourceService');
+  });
+
```

Comment:
There's no test for allocate or deallocate?

Reply:
Added, thanks.